### PR TITLE
[Local GC] Forbid inclusion of gcscan.h from VM and DAC directories

### PR DESCRIPTION
**Pull request by swgillespie**
This change removes the inclusion of `gcscan.h` from three places in the VM and once place in the DAC. For the VM uses, `IGCHeap::RuntimeStructuresValid` was added so that the VM can ask the GC whether structures are in valid state (previously done through `GCScan::GetGcRuntimeStructuresValid`). For the DAC use, the DAC variable that `GCScan::GetGcRuntimeStructuresValid` reads is referenced directly (`gc_structures_invalid_cnt`).

### CODE REVIEW

**Comment by swgillespie** [comment]
x86 job failed with https://github.com/dotnet/coreclr/issues/10022

@dotnet-bot test Windows_NT x86 Checked Build and Test

**Comment by swgillespie** [comment]
@Maoni0 @jkotas PTAL? Most of the work to enable this was done as a part of https://github.com/dotnet/coreclr/pull/9255 - so the functionality of `GCScan::GetGcRuntimeStructuresValid` can be implemented on the DAC side by reading one of the globals on the GC DAC struct.

File: src/vm/object.cpp [hunk 1]
```diff
@@ -1776,9 +1775,10 @@ VOID Object::ValidateInner(BOOL bDeep, BOOL bVerifyNextHeader, BOOL bVerifySyncB
         // try to validate next object's header
         if (bDeep 
             && bVerifyNextHeader 
-            && GCScan::GetGcRuntimeStructuresValid ()
+            && GCHeapUtilities::IsGCHeapInitialized()
```
Comment (jkotas) [inline src/vm/object.cpp]: IsGCHeapInitialized call does not look necessary - if we have an object to validate, it sure must have come from GCHeap and thus the GCHeap is initialized.

Comment (swgillespie) [inline src/vm/object.cpp]: ok!

**Comment by swgillespie** [comment]
@dotnet-bot test Linux ARM Emulator Cross Debug Build
@dotnet-bot test Linux ARM Emulator Cross Release Build

**Comment by swgillespie** [comment]
thanks!

**Pull request merged**